### ISSUE TITLE
[PSPE] Cannot pour water or lava from a bucket

### ISSUE THREAD

**Issue by magicus** [ProtocolSupport/ProtocolSupportPocketStuff#32]
On PE, it is not possible to pour water or lava from a bucket. Nothing happens when you try.

**Comment by 7kasper** [comment]
I hope thisnis related to not being able to place boat, although I dont know how to fix that yet.

**Comment by 7kasper** [comment]
I can place in creative mode although the items are remapped weirdly when I click on them. Perhaps maybe there is a slight tiny chance this is realted to a remapping problem.

**Comment by magicus** [comment]
Yeah, I see some kind of remapping issue too, now. When I select a bucket of water, it will be replaced as a bucket of lava in my hotbar after a short while. The same goes for an empty bucket. This behaviour was not there when I tested last week. I was then able to user a water bucket.

If I use it, I see a lava source block for a short instant, but then it is removed.

**Comment by magicus** [comment]
There is definitely improvements here, although not all the way to the end.

I can now actually select a bucket of water, and get it, instead of having it replaced with lava.

However, when I tried to pour it, I got the "Hot stuff" (or whatever) advancement, which sounds like the game still thought, somehow, that I was pouring lava.

I managed to get water out of the bucket. Sometimes. Most of the time, nothing happened. Sometimes, water appeared. Most of the time, the water did not float correctly. It could just stay in a block on the top of the cliff, or flow to just a single block below when I'd explected it to cascade everywhere.

**Comment by 7kasper** [comment]
Very weird. Ill look into this later this day. I suspect the ghost water might be PE guessing again and not appear to other players.

**Comment by magicus** [comment]
That was actually my guess as well. I was about to confirm this with my PC player but I was too tired to do that now. Might be able to check later this weekend.

**Comment by 7kasper** [comment]
Many of the bucket related issues should be fixed.
If you are still having issues with placing / picking up water it is because the server uses a trace from where you are looking. It's important to look straight at the water then you should be able to pick it up.

**Comment by magicus** [comment]
I'm trying to look straight and place water or lava, but no success. :( I am also unable to pick up water in an empty bucket.

**Comment by magicus** [comment]
I notice that for a moment the lava shows up on the block where I pour it, the bucket shows up as empty on the hotbar, and

`Inv transaction read: sId: 0 wId: 0 action: 0 slot: 5 oldItem: 1xitem.bucketLava@0null newItem: 1xitem.bucket@0null`

is printed to the console. But then the bucket refills, and the lava block disappear.

Don't know if this is helpful.

**Comment by magicus** [comment]
Ha, I think i pinpointed the problem!

To my suprise, I managed to pour some lava. I tried to reproduce what I did, and finally found it. I pressed long enough, to make a breaking block action. Then water and/or lava was poured just perfectly.

So there seems to be a mismatch between the two action modes, placing vs braking. Buckets should be activated the same way you place blocks, not the same way you break blocks.

Maybe an issue since in PC left click means either attack or destroy, while short press in PE means either attack or place block.
